### PR TITLE
Update MetaMask Connector documentation

### PR DESCRIPTION
**Pull request by cxalem**
This PR improves the MetaMask connector functionality by implementing the following updates:

1. Enhanced Documentation:

- Added detailed explanations for advanced features such as headless mode, logging, and dappMetadata.
- Clarified the behavior of the connector when detecting an EIP-6963 MetaMask injected provider and how it is replaced to enhance functionality.
- Provided additional context about the rdns property and its role in uniquely identifying connectors.

2. Improved Configuration Options:

- Included headless mode support for custom modals, particularly useful in web-only setups using Wagmi.
- Highlighted the use of SDKLoggingOptions for debugging RPC methods and events.

3. Chain RPC Mapping:

- Improved documentation for configuring RPC transports, ensuring better developer clarity.

4. Developer Guidance:

- Expanded on how metadata is used for UX enhancements in confirmation screens.
- Added notes on how developers can leverage advanced features to customize their dApps.

### CODE REVIEW

File: site/shared/connectors/metaMask.md [hunk 1]
```diff
@@ -43,8 +43,11 @@ Check out the [MetaMask SDK docs](https://docs.metamask.io/wallet/connect/3rd-pa
 
 `DappMetadata | undefined`
```
Comment (alexandratran) [inline site/shared/connectors/metaMask.md:44]:
```suggestion
```

Maybe we can remove these data types if it is confusing? We already mention the fields and default value in the description and provide an example.

File: site/shared/connectors/metaMask.md [hunk 2]
```diff
@@ -62,14 +65,40 @@ const connector = metaMask({
 
 `SDKLoggingOptions | undefined`
```
Comment (alexandratran) [inline site/shared/connectors/metaMask.md:67]:
```suggestion
```

File: site/shared/connectors/metaMask.md [hunk 3]
```diff
@@ -43,8 +43,11 @@ Check out the [MetaMask SDK docs](https://docs.metamask.io/wallet/connect/3rd-pa
 
 `DappMetadata | undefined`
 
-Metadata about the dapp using the SDK, including, `name`, `url`, and `iconUrl`.
-By default, `url` gets its value from `window.location.origin`.
+Metadata is used to fill details for the UX on confirmation screens in MetaMask, including the following fields:
+
+- `name`: The name of the dapp.
+- `url`: URL of the dApp (defaults to window.location.origin).
+- `iconUrl`: URL to the dApp's favicon or icon.
```
Comment (alexandratran) [inline site/shared/connectors/metaMask.md:50]:
```suggestion
- `name`: `string` - The name of the dapp.
- `url`: `string` - URL of the dapp (defaults to `window.location.origin`).
- `iconUrl`: `string` - URL to the dapp's favicon or icon.
```

File: site/shared/connectors/metaMask.md [hunk 4]
```diff
@@ -62,14 +65,40 @@ const connector = metaMask({
...
+import { metaMask } from '{{connectorsPackageName}}'
+
+const connector = metaMask({
+  logging: { developerMode: true, sdk: true } // [!code focus]
+})
+```
 
 ### headless
 
 `boolean | undefined`
```
Comment (alexandratran) [inline site/shared/connectors/metaMask.md:95]:
```suggestion
```

File: site/shared/connectors/metaMask.md [hunk 5]
```diff
@@ -85,4 +114,9 @@ const connector = metaMask({
 ## Advanced
 
 By default, if the EIP-6963 MetaMask injected provider is detected, this connector will replace it.
+
+EIP-6963 defines a standard way for dApps to interact with multiple wallets simultaneously by injecting providers into the browser. Wallets that implement this standard can make their presence known to dApps in a consistent and predictable manner.
+
+When the MetaMask SDK detects an EIP-6963-compliant provider (such as MetaMask itself), the connector will automatically replace the default injected provider (like `window.ethereum`) with the one provided by the MetaMask SDK.
```
Comment (alexandratran) [inline site/shared/connectors/metaMask.md:120]:
```suggestion
EIP-6963 defines a standard way for dapps to interact with multiple wallets simultaneously by injecting providers into the browser. Wallets that implement this standard can make their presence known to dapps in a consistent and predictable manner.

When MetaMask SDK detects an EIP-6963-compliant provider (such as MetaMask itself), the connector will automatically replace the default injected provider (like `window.ethereum`) with the one provided by MetaMask SDK.
```

File: site/shared/connectors/metaMask.md [hunk 6]
```diff
@@ -41,10 +41,11 @@ Check out the [MetaMask SDK docs](https://docs.metamask.io/wallet/connect/3rd-pa
 
 ### dappMetadata
 
-`DappMetadata | undefined`
```
Comment (tmm) [inline site/shared/connectors/metaMask.md:44]: Let's keep these types in.

File: site/shared/connectors/metaMask.md [hunk 7]
```diff
@@ -60,16 +61,38 @@ const connector = metaMask({
 
 ### logging
 
-`SDKLoggingOptions | undefined`
```
Comment (tmm) [inline site/shared/connectors/metaMask.md:63]: Let's keep these types in.

File: site/shared/connectors/metaMask.md [hunk 8]
```diff
@@ -60,16 +61,38 @@ const connector = metaMask({
...
+import { metaMask } from '{{connectorsPackageName}}'
+
+const connector = metaMask({
+  logging: { developerMode: true, sdk: true } // [!code focus]
+})
+```
 
 ### headless
 
-`boolean | undefined`
```
Comment (tmm) [inline site/shared/connectors/metaMask.md:69]: Let's keep these types in.

**Review by tmm — APPROVED** [review]

**Pull request merged**